lpcompiler.pas

```diff
@@ -2561,6 +2578,28 @@ function TLapeCompiler.ParseExpression(ReturnOn: EParserTokenSet = []; FirstNext
     end;
   end;
 
+  function __IsProperty(typ: TLapeType): Boolean;
+  begin
+    Result := (Typ <> nil) and (Typ is TLapeType_OverloadedMethod) and (TLapeType_OverloadedMethod(Typ).MethodDef = mdProperty);
+  end;
+
+  function IsProperty(Node: TLapeTree_ExprBase): Boolean;
+  begin
+    Result := False;
+    if TLapeTree_Base.isEmpty(Node) or (not (Node is TLapeTree_ExprBase)) or (Node is TLapeTree_Invoke) then
+      Exit();
+
+    if __IsProperty(Node.resType()) then
+      Exit(True);
+
+    if (Node is TLapeTree_Operator) then
```

Comment:
when autoinvoke isn't enabled and we get an expression, like:
> `typ.Prop = 10` 
this case will be registered all the way down at line 2870:
> `Result := TLapeTree_ExprBase(ResolveMethods(VarStack.Pop(), False));`
For other scenarios it doesn't have to be recursive as it's picked up earlier, or simply differently, as far as I could tell.